### PR TITLE
[AIRFLOW-7004][WIP] Lazy initialize settings

### PR DESCRIPTION
**Pull request by mik-laj**
Before:
```
15:01 $ seq 1 10 | xargs -n 1 -I {} time python -c "import airflow"
        4.93 real         1.08 user         0.29 sys
        0.96 real         0.77 user         0.17 sys
        0.96 real         0.77 user         0.16 sys
        1.02 real         0.81 user         0.17 sys
        0.96 real         0.77 user         0.16 sys
        0.99 real         0.79 user         0.17 sys
        0.99 real         0.79 user         0.17 sys
        1.06 real         0.84 user         0.18 sys
        1.00 real         0.80 user         0.16 sys
        0.97 real         0.78 user         0.16 sys
```
After:
```
15:01 $ seq 1 10 | xargs -n 1 -I {} time python -c "import airflow"
        0.12 real         0.06 user         0.02 sys
        0.07 real         0.05 user         0.01 sys
        0.07 real         0.05 user         0.01 sys
        0.08 real         0.05 user         0.01 sys
        0.10 real         0.07 user         0.02 sys
        0.10 real         0.07 user         0.02 sys
        0.13 real         0.08 user         0.02 sys
        0.12 real         0.08 user         0.02 sys
        0.11 real         0.07 user         0.02 sys
        0.09 real         0.06 user         0.02 sys
```
---
Issue link: [AIRFLOW-7004](https://issues.apache.org/jira/browse/AIRFLOW-7004)

Make sure to mark the boxes below before creating PR: [x]

- [X] Description above provides context of the change
- [X] Commit message/PR title starts with `[AIRFLOW-NNNN]`. AIRFLOW-NNNN = JIRA ID<sup>*</sup>
- [X] Unit tests coverage for changes (not needed for documentation changes)
- [X] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [X] Relevant documentation is updated including usage instructions.
- [X] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

<sup>*</sup> For document-only changes commit message can start with `[AIRFLOW-XXXX]`.

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]
Finally :)

**Review by ad-m — APPROVED** [review]

**Comment by ad-m** [comment]
@mik-laj , the result is pretty interesting. Is there anything how I could help you to merge that PR?

File: airflow/__main__.py [hunk 1]
```diff
@@ -27,13 +26,16 @@
 from airflow.cli import cli_parser
```
Comment (ad-m) [inline airflow/__main__.py:27]:
```suggestion
from airflow import settings
from airflow.cli import cli_parser
```

File: airflow/__main__.py [hunk 2]
```diff
@@ -27,13 +26,16 @@
...
 from airflow.configuration import conf
 
+from airflow import settings
```
Comment (ad-m) [inline airflow/__main__.py]:
```suggestion
```

**Review by ad-m — CHANGES_REQUESTED** [review]
Static checks fails. Following changes was suggested by isort.

**Pull request closed without merging**